data_structures/src/chain/priority.rs

```diff
@@ -0,0 +1,1169 @@
+            TimeToBlock::Exactly(x) => Some(x * seconds_per_epoch as u32),
+            TimeToBlock::LessThan(x) => Some(x * seconds_per_epoch as u32),
+            TimeToBlock::Unknown => None,
+            TimeToBlock::UpTo(x) => Some(x * seconds_per_epoch as u32),
+        }
+    }
+
+    /// Convert a `TimeToBlock` into a formatted string expressing the time-to-block in seconds.
+    ///
+    /// This requires knowledge of the number of seconds per epoch, aka __checkpoint period__.
+    pub fn pretty_print_secs(&self, seconds_per_epoch: u16) -> String {
+        fn unit_change(seconds: u32, divider: u32) -> (u32, u32, String) {
+            let value = seconds / divider;
+            let remainder = seconds % divider;
+            let plural = String::from(if value == 1 { "" } else { "s" });
+
+            (value, remainder, plural)
+        }
+
+        let string = self
+            .as_secs(seconds_per_epoch)
+            .map(|seconds| {
+                let mut strings = Vec::<String>::new();
+                let (days, remainder, plural) = unit_change(seconds, 24 * 60 * 60);
+                if days > 0 {
+                    strings.push(format!("{} day{}", days, plural))
+                }
+                let (hours, remainder, plural) = unit_change(remainder, 60 * 60);
+                if hours > 0 {
+                    let separator = if strings.is_empty() {
+                        ""
+                    } else if remainder > 0 {
+                        ", "
+                    } else {
+                        " and "
+                    };
+
+                    strings.push(format!("{}{} hour{}", separator, hours, plural))
+                }
+                let (minutes, remainder, plural) = unit_change(remainder, 60);
+                if minutes > 0 {
+                    let separator = if strings.is_empty() {
+                        ""
+                    } else if remainder > 0 {
+                        ", "
+                    } else {
+                        " and "
+                    };
+                    strings.push(format!("{}{} minute{}", separator, minutes, plural))
+                }
+                let (seconds, _, plural) = unit_change(remainder, 1);
+                if seconds > 0 || (minutes == 0 && hours == 0 && days == 0) {
+                    let separator = if strings.is_empty() { "" } else { " and " };
+                    strings.push(format!("{}{} second{}", separator, seconds, plural));
+                }
+
+                strings.join("")
+            })
+            .unwrap_or_default();
+
+        match self {
+            TimeToBlock::Around(_) => format!("around {}", string),
+            TimeToBlock::Exactly(_) => string,
+            TimeToBlock::LessThan(_) => format!("less than {}", string),
+            TimeToBlock::Unknown => String::from("unknown"),
+            TimeToBlock::UpTo(_) => format!("up to {}", string),
+        }
```

Comment:
I don't understand the difference between all the options, when I see the priorities in the CLI output I just assume that everything is "around", because it is just an estimate so even if it says "less than 1 minute 30 seconds" I assume it means "around 1 minute 30 seconds".

Reply:
Got rid of this madness.